containers/ecr-viewer/environment.d.ts

```diff
@@ -1,18 +1,133 @@
 /* eslint-disable unused-imports/no-unused-vars */
-namespace NodeJS {
-  interface ProcessEnv {
-    APP_VERSION: string; // Version of eCR Viewer. This value is set at build time
+namespace EnvironmentVariables {
+  /**
+   * @categoryDescription Authentication - Integrated
+   * These variables are used to configure authentication for the eCR Viewer.
+   * All values are required when running in `INTEGRATED` or `DUAL`.
+   * @categoryDescription Authentication - Non-Integrated
+   * These variables are used to configure authentication for the eCR Library & Viewer.
+   * All values are required when running in `NON_INTEGRATED` or `DUAL`.
+   */
+  interface Authentication {
+    //#region auth_non_integrated
+    /**
+     * @category Authentication - Non-Integrated
+     * @description The application/client id used to identify the client.
+     */
     AUTH_CLIENT_ID?: string;
+    /**
+     * @category Authentication - Non-Integrated
+     * @description The client secret that comes from the authentication provider.
+     */
     AUTH_CLIENT_SECRET?: string;
+    /**
+     * @category Authentication - Non-Integrated
+     * @description Additional information used during authentication process. For Azure AD, this will be the 'Tenant Id'. For Keycloak, this will be the url issuer including the realm.
+     * @example Keycloak
+     * https://my-keycloak-domain.com/realms/My_Realm
+     * @example Azure
+     * 4a62fd3e-be14-443f-b51d-a9facca4a0eb
+     */
     AUTH_ISSUER?: string;
+    /**
+     * @category Authentication - Non-Integrated
+     * @description The authentication provider used for logging in. Supported values are `keycloak` for Keycloak or `ad` for Azure AD/Entra.
+     */
     AUTH_PROVIDER?: "keycloak" | "ad";
+    /**
+     * @category Authentication - Non-Integrated
+     * @description A random key used for NextAuth.js sessions. This value can be generated by running `openssl rand -base64 32`.
+     */
+    NEXTAUTH_SECRET: string;
+    //#endregion auth_non_integrated
+    //#region auth_integrated
+    /**
+     * @category Authentication - Integrated
+     * @description Public key for NBS authentication.
```

Comment:
I don't think we have much information about how to set this on the NBS side - could we add some details here? Thinking something like:

> This public key should correspond to a private key that you generate and store within the NBS Configuration table. To generate these keys, you can use your tool of choice [link whatever folks are using these days]